### PR TITLE
Error for not yet implemented strokeCap in WEBGL 

### LINKED ISSUE #3410: strokeCap() not defined in WEBGL
Ideally this would be implemented, but at very least should output a 'not-yet-implemented' warning rather than crash the sketch.

#### Nature of issue?

- [x] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [x] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [x] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: 0.72
- Web browser and version: current chrome and firefox
- Operating System: OS X 10.13.6
- Steps to reproduce this:
1. Go to this page: https://p5js.org/examples/data-variables.html
2. Change  createCanvas(720, 400) to  createCanvas(720, 400, WEBGL);
3. Sketch crashes with following error in console

```
p5.RendererGL: enabled webgl context p5.min.js:14:17029
TypeError: this._renderer.strokeCap is not a function[Learn More] p5.min.js:9:12731
[31]</d.prototype.strokeCap
https://p5js.org/assets/js/p5.min.js:9:12731
<anonymous> self-hosted:974:17 setup
https://p5js.org/assets/examples/example.html:7:3
e/this._setup<
https://p5js.org/assets/js/p5.min.js:8:8048
<anonymous> self-hosted:972:17 e/this._start<
https://p5js.org/assets/js/p5.min.js:8:7032
<anonymous> self-hosted:972:17 e
https://p5js.org/assets/js/p5.min.js:8:10541
<anonymous>
https://p5js.org/assets/examples/example.html:34:1
loadExample
https://p5js.org/assets/js/examples.js:174:57
init/<
https://p5js.org/assets/js/examples.js:45:11
dispatch
https://p5js.org/assets/js/vendor/jquery-1.12.4.min.js:3:12392
add/r.handle
https://p5js.org/assets/js/vendor/jquery-1.12.4.min.js:3:9156

```

### PR DESCRIPTION
**Pull request by sanketsingh24**
Fixes #3410 

- Implemented an error for strokeCap() in webgl mode.
- Also removed _TODO clear_, which i found to be implemented.

### CODE REVIEW

File: src/webgl/p5.RendererGL.js [hunk 1]
```diff
@@ -417,11 +417,6 @@ p5.RendererGL.prototype.background = function() {
   this.GL.clear(this.GL.COLOR_BUFFER_BIT | this.GL.DEPTH_BUFFER_BIT);
 };
 
-//@TODO implement this
-// p5.RendererGL.prototype.clear = function() {
```
Comment (sanketsingh24) [inline src/webgl/p5.RendererGL.js]: line number 716

**Comment by sanketsingh24** [comment]
Actually, I just went through RendererGL.js, and found two instances ( [here](https://github.com/processing/p5.js/blob/7ddfa32cb44fcb528a853fc66090ba1202dfb09e/src/webgl/p5.RendererGL.js#L161) and [here](https://github.com/processing/p5.js/blob/7ddfa32cb44fcb528a853fc66090ba1202dfb09e/src/webgl/p5.RendererGL.js#L1118)) to be using `throw new Error`, so I used it. But now when I searched the whole project, `console.error` is also present. 

In my opinion, `console.error` is best for this use case. But `throw new Error` also gives out visual output in the console, along with the error type and the call stack, while console.error just prints out whatever given.

**Comment by stalgiag** [comment]
@sanketsingh24 arguments definitely could definitely be made in favor of `throw new Error` I think in situations such as failure to create a rendering context it makes more sense to break the drawing and show the call stack. Then again it seems that _assert3D will also break rendering if 3D functions are used in a 2D sketch, so I don't really feel that there are hard and fast rules here. My personal opinion is that `strokeCap` shouldn't break the drawing simply because it would be rare for `strokeCap` to be integral to a sketch.

**Comment by sanketsingh24** [comment]
@stalgiag That was what exactly I was thinking, and I've pushed the changes :smile:

**Comment by stalgiag** [comment]
Thanks @sanketsingh24!

**Pull request merged**